src/shared/tpm2-util.c

```diff
@@ -458,24 +458,161 @@ static int tpm2_make_primary(
         return 0;
 }
 
-void tpm2_pcr_mask_to_selection(uint32_t mask, uint16_t bank, TPML_PCR_SELECTION *ret) {
-        assert(ret);
+/* Utility functions for TPMS_PCR_SELECTION. */
+
+static uint32_t tpm2_tpms_pcr_selection_to_mask(const TPMS_PCR_SELECTION *s) {
+        uint32_t mask = 0;
+        for (unsigned i = 0; i < s->sizeofSelect; i++)
+                SET_FLAG(mask, s->pcrSelect[i] << (i * 8), true);
+        return mask;
+}
+
+static TPMS_PCR_SELECTION tpm2_tpms_pcr_selection_from_mask(uint32_t mask, TPMI_ALG_HASH hash) {
```

Comment:
hmm, how large is the struct we return here?

returning structs is a bit icky, in particular if it has more than a few fields. This struct is kinda large, no?

i mean, there's a reason why tpm2_pcr_mask_to_selection() passed this in via a ptr...

Reply:
I changed this to an inline function in the header; I think that should avoid actually returning a large struct, does it look ok? I can change it to take a ptr param instead of returning the struct value if you think that's better, it just seems like that way is more work for the caller and (IMHO) less 'natural' way to use it.